### PR TITLE
Fixed a bug to create stateTrieMigrationDB initial folder

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes
There is a bug when creating `statetrie_` database directory in the last PR.
This PR change not to create the `statetrie_` folder.

## Types of changes

Please put an x in the boxes related to your change.

- [x] Bugfix
- [ ] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Related issues

- Please leave the issue numbers or links related to this PR here.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by KimKyungup** [comment]
PTAL. :)

File: storage/database/db_manager.go [hunk 1]
```diff
@@ -362,12 +362,12 @@ func partitionedDatabaseDBManager(dbc *DBConfig) (*databaseManager, error) {
 
 		if entryType == StateTrieDB || entryType == StateTrieMigrationDB {
 			dir := dbm.getDBDir(entryType)
-			newDBC := getDBEntryConfig(dbc, entryType, dir)
-			if entryType == StateTrieMigrationDB && dir == "" {
+			if entryType == StateTrieMigrationDB && dir == dbDirs[StateTrieMigrationDB] {
```
Comment (winnie-byun) [inline storage/database/db_manager.go:365]:
```suggestion
			if entryType == StateTrieMigrationDB {
```
Is `dir == dbDirs[StateTrieMigrationDB]` necessary?

Comment (KimKyungup) [reply storage/database/db_manager.go:365]: Yes it is. For restarting state trie migration, `StateTrieMigrationDB` should be loaded `statetrie_N`.
But if there is no restart job, it doesn't need to create `statetrie_` database.

File: storage/database/db_manager.go [hunk 2]
```diff
@@ -477,7 +477,11 @@ func (dbm *databaseManager) NewBatch(dbEntryType DBEntryType) Batch {
 
 func (dbm *databaseManager) getDBDir(dbEntry DBEntryType) string {
 	miscDB := dbm.getDatabase(MiscDB)
+
 	enc, _ := miscDB.Get(databaseDirKey(uint64(dbEntry)))
+	if len(enc) == 0 {
+		return dbDirs[dbEntry]
+	}
```
Comment (aidan-kwon) [inline storage/database/db_manager.go:484]: I think we need to check wheter `enc` is `nil` or not. 
Or we can check and print an error message when `miscDB.Get()` failed.
```
enc, _ := miscDB.Get(databaseDirKey(uint64(dbEntry)))
if enc != nil && len(enc) != 0 {
	return string(enc)
}
return dbDirs[dbEntry]
```

File: storage/database/db_manager.go [hunk 3]
```diff
@@ -477,7 +477,11 @@ func (dbm *databaseManager) NewBatch(dbEntryType DBEntryType) Batch {
 
 func (dbm *databaseManager) getDBDir(dbEntry DBEntryType) string {
 	miscDB := dbm.getDatabase(MiscDB)
+
 	enc, _ := miscDB.Get(databaseDirKey(uint64(dbEntry)))
+	if len(enc) == 0 {
+		return dbDirs[dbEntry]
```
Comment (aidan-kwon) [inline storage/database/db_manager.go:483]: As I mentioned on the face to face discussion, we may need to change the name of `dbDirs` to `dbBaseDirs` or something else.

File: storage/database/db_manager.go (hunk 2 above)
Comment (winnie-byun) [reply storage/database/db_manager.go:484]: Checking error and logging the message sounds good.
However,  I think @aidan-kwon 's suggestion will make a difference. `len(enc) == 0` returns true for both when enc is nil and empty byte.

File: storage/database/db_manager.go [hunk 4]
```diff
@@ -362,12 +362,12 @@ func partitionedDatabaseDBManager(dbc *DBConfig) (*databaseManager, error) {
 
 		if entryType == StateTrieDB || entryType == StateTrieMigrationDB {
```
[the remark this answers is not in the document]
Comment (KimKyungup) [reply storage/database/db_manager.go:363]: For example?

File: storage/database/db_manager.go (hunk 2 above)
Comment (aidan-kwon) [reply storage/database/db_manager.go:484]: @winnie-byun You're right. It was my mistake. 
I will resolve this conversation.

File: storage/database/db_manager.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (winnie-byun) [reply storage/database/db_manager.go:363]:
```
switch entryType {
		case StateTrieDB:
			dir := dbm.getDBDir(entryType)
			newDBC := getDBEntryConfig(dbc, entryType, dir)
			if dbc.NumStateTriePartitions > 1 {
				db, err = newPartitionedDB(newDBC, entryType, dbc.NumStateTriePartitions)
			} else {
				db, err = newDatabase(newDBC, entryType)
			}
		case StateTrieMigrationDB:
			dir := dbm.getDBDir(entryType)
			if dir == dbDirs[StateTrieMigrationDB] {
				// If there is no migration DB, skip to set.
				continue
			}
		default:
			newDBC := getDBEntryConfig(dbc, entryType, dbDirs[entryType])
			db, err = newDatabase(newDBC, entryType)
		}
```
Like this.

File: storage/database/db_manager.go (hunk 3 above)
Comment (KimKyungup) [reply storage/database/db_manager.go]: Reflected

File: storage/database/db_manager.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (KimKyungup) [reply storage/database/db_manager.go]: Reflected

**Pull request merged**